tensorrt_llm/_torch/modules/fused_moe.py

```diff
@@ -565,6 +569,12 @@ def forward_chunk(
         assert token_final_scales.dtype == torch.float32
         assert token_selected_experts.dtype == torch.int32
 
+        if self.apply_router_weight_on_input:
+            assert self.routing_method.top_k == 1, "Current walkaround only supports top-1 routing"
+            x = x * token_final_scales.to(x.dtype)
```

Comment:
```suggestion
            x = x * token_final_scales
```

Reply:
It looks like type promotion would promote the input (`x`) to float, matching the dtype of `token_final_scales`.